cmd/zfs/zfs_main.c

```diff
@@ -7879,6 +7884,98 @@ zfs_do_diff(int argc, char **argv)
 	return (err != 0);
 }
 
```

Comment:
The scrub_single_file() return code is a little non-standard here:
0: Fails stat check
1: Success
2: Scrub error

I would do:
0: Success
1: Fails stat check
2: Scrub error

Reply:
Nevermind - if these return codes are needed for `zfs_iter_filesystems_v2(zhp, 0, find_dataset_and_scrub, data)` then just ignore me.

Reply:
The reason for this is how the `zfs_iter_root`/` zfs_iter_filesystems_v2` works returning != 0 means stop traverse. In case when, we `stat` fails we want to continue traverse same as we are unable to get montpoint, however, if we were able to find a file/start scrubbing you want to indicate error.

Reply:
Oh, sorry I didn't see you answer :)